client/src/helpers/mutationOptions.ts

```diff
@@ -0,0 +1,34 @@
+import { getUpdateFunction, CacheOperation } from 'offix-cache';
+import { GET_TASKS } from '../gql/queries';
+
+
+const options = {
+  updateQuery: GET_TASKS,
+  returnType: 'Task',
+};
+
+export const mutationOptions = {
```

Comment:
There's quite a bit of obfuscation here that doesn't really bring much benefit. These global cache updates are already one of the most confusing parts of offix.

I have two comments

1. I'd get rid of the `options` object and duplicate those properties on each object. (duplication is not always the enemy) This way it's immediately more obvious what properties are needed for `getUpdateFunction`
2. The usage of the `add` `edit` and `remove` keys are potentially confusing. As a new user of Offix, I might think that these are somehow significant to the API.

I'd prefer to see something like this:

```ts
const addTask = { //or something like addTaskOptions
  updateQuery: GET_TASKS,
  returnType: 'Task',
  mutationName: 'createTask',
  operationType: CacheOperation.ADD,
}

const updateTask = {
  updateQuery: GET_TASKS,
  returnType: 'Task',
  mutationName: 'updateTask',
  operationType: CacheOperation.REFRESH,
}

// and so on

// this could also be called `globalCacheUpdates`
// because it's a term we use in the docs
// and when speaking to each other as teammates and other community members
// it's important that we build a shared understanding of specific terminology we use
export const taskCacheUpdates = {
  addTask: getUpdateFunction(addTask),
  updateTask: getUpdateFunction(updateTask),
  // and so on
}
```

Reply:
Cool, thanks for that insight. I agree completely with the comments on building and sharing the specific terminology. It also made me realise that I'm using the global cache updates, but then using them locally in each mutation function, so I will go and clean those mutations now too.